src/targets/gpu/clip.cpp

```diff
@@ -0,0 +1,26 @@
+#include <migraphx/gpu/clip.hpp>
+#include <migraphx/gpu/context.hpp>
+#include <migraphx/gpu/device/clip.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+shape hip_clip::compute_shape(std::vector<shape> inputs) const
+{
+    inputs.pop_back();
+    return op.compute_shape(inputs);
+}
+
+argument
+hip_clip::compute(context& ctx, const shape& output_shape, const std::vector<argument>& args) const
+{
+    assert(output_shape == args.back().get_shape());
```

Comment:
What is the reason for this assert?

Reply:
This was copied from another compute function, so I don't know if it's necessary.

Reply:
Well if the allocation is different we now have the `adjust_gpu_allocation` pass that will fix these allocations.